### DISCUSSION TITLE
Uniform four-digit numbers?

### DISCUSSION THREAD

**Discussion by thejohnfreeman** [XRPLF/XRPL-Standards#32] [Meta]
Excluding leading zeroes from specification numbers leads to "bad" ordering in most filesystems, and on GitHub. I propose we use four-digit numbers with leading zeroes for all spec numbers (in filenames and URIs at least), just like PEPs and RFCs. I suspect we won't hit 10,000 proposals any time soon.

**Comment by Wietse-Livingroom** [comment]
Yes, this makes a lot of sense imo.
@alloynetworks @mDuo13 @RichardAH @RareData

**Comment by WietseWind** [comment]
OK, I'm going to update the existing proposals 👍

**Comment by RichardAH** [comment]
As I've done in this standard proposal [0015. XLS15d: Concise Transaction Identifier](https://github.com/XRPLF/XRPL-Standards/discussions/34) we can prefix with the left 0 padded number without losing the 'canonical name' of the standard which is easy to reference and human readable, e.g. `xls15`

[a comment by an automated account is not shown]
